### ISSUE TITLE
Latest Sundials does not work on Julia v1.11+

### ISSUE THREAD

**Issue by staticfloat** [SciML/Sundials.jl#441]
_**Describe_ the bug 🐞**

The [SuiteSparse version bump](https://github.com/JuliaLang/julia/commit/c5dcf4f56d9e51ca0a83f5e320377d28e4d4b11a) on Julia v1.11 is no longer compatible with `Sundials_jll`.  This breaks installation of the latest version of `Sundials.jl` on Julia `master`.

**Expected behavior**

A new version of `Sundials_jll` should be compiled that works with SuiteSparse v7.4.0.

**Minimal Reproducible Example 👇**

```julia
(tmp.jQGX3ZJqve) pkg> add Sundials_jll@5.2.2
   Resolving package versions...
ERROR: Unsatisfiable requirements detected for package SuiteSparse_jll [bea87d4a]:
 SuiteSparse_jll [bea87d4a] log:
 ├─possible versions are: 7.4.0 or uninstalled
 └─found to have no compatible versions left with Sundials_jll [fb77eaff]
   └─Sundials_jll [fb77eaff] log:
     ├─possible versions are: 3.1.1 - 6.6.0 or uninstalled
     └─restricted to versions 5.2.2 by an explicit requirement, leaving only versions: 5.2.2
```

  - Output of `versioninfo()`

```julia
julia> versioninfo()
Julia Version 1.11.0-DEV.1290
Commit 8c49e5e476* (2024-01-18 14:41 UTC)
Platform Info:
  OS: macOS (arm64-apple-darwin23.1.0)
  CPU: 10 × Apple M1 Pro
  WORD_SIZE: 64
  LLVM: libLLVM-16.0.6 (ORCJIT, apple-m1)
Threads: 1 default, 0 interactive, 1 GC (on 8 virtual cores)
```

**Comment by ViralBShah** [comment]
This fixes Sundials 5 to SuiteSparse 7.2, which can be relaxed: https://github.com/JuliaPackaging/Yggdrasil/blob/619f01b5b1bbba6dab633bea6eb8782934232102/S/Sundials/Sundials%405/build_tarballs.jl#L94

**Comment by ViralBShah** [comment]
New Sundials_jll 5.2.3 is now published.
